tests/access_test.go

```diff
@@ -0,0 +1,64 @@
+package tests
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	authorization "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	commonTemplates "kubevirt.io/ssp-operator/internal/operands/common-templates"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("[rfe_id:500]Golden Images RBAC", func() {
+	BeforeEach(func() {
+		namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountNS}}
+		Expect(apiClient.Create(ctx, namespaceObj)).ToNot(HaveOccurred(), "Failed to create namespace")
+
+		saObj := &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountName, Namespace: testServiceAccountNS}}
+		Expect(apiClient.Create(ctx, saObj)).ToNot(HaveOccurred(), "Failed to create ServiceAccount")
+
+		waitUntilDeployed()
+	})
+	AfterEach(func() {
+		saObj := &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountName, Namespace: testServiceAccountNS}}
+		err := apiClient.Delete(ctx, saObj)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		namespaceObj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testServiceAccountNS}}
+		err = apiClient.Delete(ctx, namespaceObj)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+		waitForDeletion(client.ObjectKey{
+			Name: testServiceAccountNS,
+		}, &v1.Namespace{})
+	})
+
+	table.DescribeTable("should verify resource permissions", func(resourceAttr *authorization.ResourceAttributes, expectedFail bool) {
+		sar := &authorization.SubjectAccessReview{
+			Spec: authorization.SubjectAccessReviewSpec{
+				User:               fmt.Sprintf("system:serviceaccount:%s:%s", testServiceAccountNS, testServiceAccountName),
+				ResourceAttributes: resourceAttr,
+			},
+		}
+		sar.Spec.ResourceAttributes.Namespace = commonTemplates.GoldenImagesNSname
+		allowed, err := CanIAccess(sar)
+		Expect(err).ToNot(HaveOccurred())
+		if expectedFail {
+			Expect(allowed).To(BeFalse(), "Expected permission to NOT be allowed")
+		} else {
+			Expect(allowed).To(BeTrue(), "Expected permission to be allowed")
+		}
+	},
+		table.Entry("[test_id:4775]: ServiceAccounts with view role can view PVCs",
+			&authorization.ResourceAttributes{Verb: "get", Resource: "pvc"}, false),
+		table.Entry("[test_id:4776]: ServiceAccounts with only view role cannot create PVCs",
+			&authorization.ResourceAttributes{Verb: "create", Resource: "pvc"}, true),
+		table.Entry("[test_id:4846]: ServiceAccounts with only view role cannot delete PVCs",
+			&authorization.ResourceAttributes{Verb: "delete", Resource: "pvc"}, true),
+		table.Entry("[test_id:4875]: ServiceAccounts with only view role cannot view DVs",
+			&authorization.ResourceAttributes{Verb: "get", Resource: "datavolume"}, false),
```

Comment:
This test is called "ServiceAccounts with only view role CANNOT view DVs", but the test code expects success.